docs(waku): adds furps for communities over Waku

jrainville opened the pull request:
Adds FURPS for how Satus communities should work over Waku

I made them assuming the description was no longer on Waku but instead of Codex (or somewhere else magically).
Let me know if you want the current version too and I can add it

jazzz commented on docs/FURPS/communities-over-waku.md:
```diff
@@ -0,0 +1,51 @@
+## Functionality
+
+- **Community messages** (channel posts, updates) must be sent and received via **Waku**.
+- **Description change notifications** must be broadcast on Waku when the community description is updated via Codex.
```
```suggestion
- Participants must be notified of **Description changes**.
```

"Broadcast on Waku" implies that that all nodes need to be notfied rather than limited to  community members.

jazzz commented on docs/FURPS/communities-over-waku.md:
```diff
@@ -0,0 +1,51 @@
+  - Temporary disconnections
+  - Slow or lossy networks
+- Waku store nodes must reliably store messages for **30 days**, and make them retrievable on demand.
+- Description change messages must be **deduplicated and verifiable**.
```
What particular properties need to be verified?

jazzz commented on docs/FURPS/communities-over-waku.md:
```diff
@@ -0,0 +1,51 @@
+- **Partial updates** (e.g., channel edits, member changes) can be proposed by **admins**, but must be sent to the **control node** for approval before being published.
+- Communities support a **permission model**:
+  - **Control node**: the sole authority for publishing official community description changes.
+  - **Token masters**: same rights as owners except cannot manage other token masters.
```
'Owners' are not defined here

jazzz commented on docs/FURPS/communities-over-waku.md:
```diff
@@ -0,0 +1,51 @@
+
+## Functionality
+
+- **Community messages** (channel posts, updates) must be sent and received via **Waku**.
```
This requirement makes sense, but out of curiosity is Waku a hard requirement here?

jrainville commented:
@fryorcraken @jazzz thank you for the comments. I think I addressed them all.

Summary: I changed the format to be agnostic of Waku and use the Logos Stack generically.

I also added an annex at the bottom to describe what the Community Description contains.

Later, after I've had a meeting with Patryk, I'll also add FURPS for when the community is on chain instead.

fryorcraken commented on docs/FURPS/communities.md:
```diff
@@ -0,0 +1,130 @@
+
+## Functionality
+
+- **Community messages** (channel posts, updates) must be sent and received via the **Messaging protocol**.
+- The Community will be described in the **Community Description**
+  - See Annex A below for the list of properties of the Community Description
+- The Community Description must be stored on the **Storage Protocol**
```
```suggestion
- **Community messages** (channel posts, updates) must be exchanged between users of the community in a real-time.
- The Community will be described in the **Community Description**
  - See Annex A below for the list of properties of the Community Description
- The Community Description must be available to all users of the community. Only the community owner can update the community description.
```

You just replaced Waku and Codex with "Messaging Protoocl or Storage Protocol". My suggestion was to focus on the actual functionality you want. I don't think you, or your users, care what kind of protocol is used.

fryorcraken reviewed (approved):
I think this is the neat-est Status requirements I have ever read :]

*by approving I am not committing to the delivery of those requirements :D but that they are well defined and enable discussions around fulfilling said requirements with Waku software*
